src/cpp/mip/mip_field.hpp

```diff
@@ -49,25 +49,28 @@ class FieldView : public C::mip_field_view
     uint8_t fieldDescriptor() const { return C::mip_field_field_descriptor(this); }
     ///@brief Returns the descriptor set and field descriptor.
     CompositeDescriptor descriptor() const { return {descriptorSet(), fieldDescriptor()}; }
-    ///@copydoc mip::C::mip_field_payload_length
-    uint8_t payloadLength() const { return C::mip_field_payload_length(this); }
+
     ///@copydoc mip::C::mip_field_total_length
     uint8_t totalLength() const { return C::mip_field_total_length(this); }
-    ///@copydoc mip::C::mip_field_payload
-    const uint8_t* payloadPointer() const { return C::mip_field_payload(this); }
+
+    ///@copydoc mip::C::mip_field_payload_length
+    uint8_t payloadLength() const { return C::mip_field_payload_length(this); }
+
+    ///@brief Get a const view of the payload data.
+    microstrain::ConstU8ArrayView payload() const { return {C::mip_field_payload(this), C::mip_field_payload_length(this)}; }
 
     ///@brief Index the payload at the given location.
     ///@param index Byte index into payload. 0 <= index < payloadLength().
     ///@returns payload byte
-    uint8_t payload(size_t index) const { assert(index < payloadLength()); return payloadPointer()[index]; }
+    uint8_t payload(size_t index) const { assert(index < payloadLength()); return payload()[index]; }
```

Comment:
This might make more sense to be called payloadByte or just byte since it's not getting the payload itself

Reply:
Renamed to `payloadAt(size_t index)` and added `dataAt(size_t index)` to match